### PR TITLE
lammps: add stable versions 20230802, 20220623.4

### PR DESCRIPTION
**Pull request by zzzoom**

### CODE REVIEW

File: var/spack/repos/builtin/packages/lammps/package.py [hunk 1]
```diff
@@ -28,17 +28,28 @@ class Lammps(CMakePackage, CudaPackage, ROCmPackage):
     #   marked deprecated=True
     # * patch releases older than a stable release should be marked deprecated=True
     version("develop", branch="develop")
+    version(
+        "20230802",
+        sha256="48dc8b0b0583689e80ea2052275acbc3e3fce89707ac557e120db5564257f7df",
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/lammps/package.py:33]: Weird.  I'm not getting this sha256 from `spack checksum`.

**Review by tldahlgren — COMMENTED** [review]
I can confirm the 20220623.4 sha256 but not 20230802.  Can you re-check?

**Comment by zzzoom** [comment]
@tldahlgren did you run `spack checksum` without applying the PR? The lammps package encodes whether it downloads a patch or a full release with preferred versions, you might have checksummed `patch_2Aug2023.tar.gz` instead of `stable_2Aug2023.tar.gz` because of that.

File: var/spack/repos/builtin/packages/lammps/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/lammps/package.py:33]: Now I can confirm this sha256 BUT not the next one.  I made sure I have this PR checked out too.

**Comment by tldahlgren** [comment]
> @tldahlgren did you run `spack checksum` without applying the PR? The lammps package encodes whether it downloads a patch or a full release with preferred versions, you might have checksummed `patch_2Aug2023.tar.gz` instead of `stable_2Aug2023.tar.gz` because of that.

It's possible I was using the wrong window/branch.  But, as I mentioned in prior comment, it looks like I can confirm still only confirm one of the sha256 (just not the one before).  Weird.

**Comment by tldahlgren** [comment]
@zzzoom Should this PR be closed since the maintainer (@rbberger) has more extensive changes in #39317 ?

**Comment by zzzoom** [comment]
Ok

**Pull request closed without merging**